src/ucp/proto/proto_common.c

```diff
@@ -97,9 +97,14 @@ size_t
 ucp_proto_common_get_max_frag(const ucp_proto_common_init_params_t *params,
                               const uct_iface_attr_t *iface_attr)
 {
+    size_t max = SIZE_MAX;
+
+    if (params->flags & UCP_PROTO_COMMON_INIT_FLAG_MAX_FRAG) {
+        max = params->max_length;
+    }
```

Comment:
instead, initialize min/max_length fields in all protocols init_params structures